internal/provider/resource_port_test.go

```diff
@@ -0,0 +1,43 @@
+package provider
+
+import (
+	"log"
+	"strconv"
+	"testing"
+
+	"github.com/PacketFabric/terraform-provider-packetfabric/internal/testutil"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccPort(t *testing.T) {
+	testutil.SkipIfEnvNotSet(t)
+
+	portDetails := testutil.CreateBasePortDetails()
+	portTestResult := portDetails.RHclPort()
+	log.Println(portTestResult.ResourceReference)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testutil.PreCheck(t, []string{})
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:             portTestResult.Hcl,
+				ExpectNonEmptyPlan: true,
```

Comment:
Why is this set to true?

Reply:
I'm not following, why is it an issue? (https://developer.hashicorp.com/terraform/plugin/sdkv2/best-practices/testing#built-in-patterns)

Reply:
> TestStep offers a boolean field [ExpectNonEmptyPlan](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestStep.ExpectNonEmptyPlan) as well [ExpectError](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestStep.ExpectError) regex field to specify ways the test framework can handle expected failures

Is this an **expected** failure here?

Reply:
@pontovinte. As I understand reading [terraform documentation](https://developer.hashicorp.com/terraform/plugin/sdkv2/best-practices/testing), we should not set this value to true unless we expect a failure.

Reply:
@rjouhann, the terraform tests are supposed to run the following: `terraform plan`, `terraform apply` and `terraform destroy`. Setting `ExpectNonEmptyPlan` to true mitigates the earlier phases of a test step (terraform plan) prior a `terraform apply`. The same thing goes when transitioning from `terraform apply` to `terraform destroy`.
I think the inline comments ([here](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestStep.ExpectNonEmptyPlan)) are more helpful to understand, rather than TF docs.

Reply:
Thanks for the additional info. I think both Adam and I would like to keep terraform plan in the terraform test logic though :)